ext/io/console/console.c

```diff
@@ -275,16 +275,17 @@ set_ttymode(int fd, conmode *t, void (*setter)(conmode *, void *), void *arg)
     return setattr(fd, &r);
 }
 
-#define GetReadFD(fptr) ((fptr)->fd)
+#define GetReadFD(fptr) rb_ioptr_descriptor(fptr)
 
 static inline int
 get_write_fd(const rb_io_t *fptr)
 {
-    VALUE wio = fptr->tied_io_for_writing;
+    VALUE wio = rb_ioptr_tied_io_for_writing(fptr);
+    if (!wio) return rb_ioptr_descriptor(fptr);
+
     rb_io_t *ofptr;
-    if (!wio) return fptr->fd;
     GetOpenFile(wio, ofptr);
-    return ofptr->fd;
+    return rb_ioptr_descriptor(ofptr);
```

Comment:
We can create compatibility macros for this case, wdyt?

Reply:
I don't think we should expose any parts of `struct rb_io`.

Reply:
Alright.
Maybe we could always define macros like `HAVE_RB_IO_DESCRIPTOR` which means `rb_ioptr_descriptor` is available without needing to `have_func('rb_ioptr_descriptor')` which is quite slow (one C compilation for each function checked).
Or something more direct like `RB_IO_DESCRIPTOR(io)`, and so users can just define it as the old way if not already defined.

I'm not a big fan of the `rb_ioptr_` naming. I think `rb_io_` is better. That might mean passing the VALUE instead of the `struct rb_io`, that seems fine and much clearer that structs are not parts of the API (even opaque structs).

Reply:
I'm okay with this but I'm between a rock and a hard place, I didn't create the interfaces that take the "fptr" argument and I'm all for defining more `rb_io_...(VALUE)` functions. I agree the less of this we expose the better.